### DISCUSSION TITLE
How do I mock Time for system tests?

### DISCUSSION THREAD

**Discussion by ostwilkens** [bevyengine/bevy#2546] [Q&A]
I need to ensure the results of my systems are framerate-independent. 
To do this, I'm thinking I need to be able to update Time with an Instant of my choosing. 
There is a function for Time named update_with_instant, but it's only public to the bevy crate.

**Comment by ManevilleF** [comment]
You can use [custom time steps](https://bevy-cheatbook.github.io/features/fixed-timestep.html)

**Reply by ostwilkens** [reply]
Thank you for the tip, I didn't know this was a built-in feature! 
However, I want the systems to run every frame, and the framerate can vary. 
I am using time delta, and I think the system is framerate-independent, but I need to confirm it.  


If update_with_instant was public, I would have everything I need. But I'm guessing it's a conscious decision to have it private.

**Comment by alice-i-cecile** [comment]
For future readers, #2549 (by the same author) exposes the relevant functionality to the world. I think this is the right direction to go; let's see how that PR process works out.
